### PR TITLE
Add complete conda env to faster solving

### PR DESCRIPTION
**Pull request by MaxUlysse**
# nf-core/sarek pull request

Many thanks for contributing to nf-core/sarek!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by MaxUlysse** [comment]
@apeltzer any idea on how to simplify the `environment.yml` file?

**Comment by apeltzer** [comment]
I've recently read something about removing detailed dependencies wherever possible as this makes the solving faster. E.g. if you have samtools which depends on htslib, then don't explicitly import htslib and let the samtools recipe take care of finding htslib and installing it for you. 

I'd try to only have the "major" packages in the environment.yaml and then try to find out whether this builds in the end then ...?

File: environment.yml [hunk 1]
```diff
@@ -6,36 +6,525 @@ channels:
...
+  - bzip2=1.0.8=h7f98852_4
+  - c-ares=1.17.1=h36c2ea0_0
+  - cairo=1.16.0=hcf35c78_1003
+  - cancerit-allelecount=4.0.2=ha228f0b_1
+  - certifi=2020.6.20=pyhd3eb1b0_3
+  - cffi=1.14.0=py27hd463f26_0
+  - chardet=3.0.4=py27h8c360ce_1006
+  - click=7.1.2=pyh9f0ad1d_0
+  - clustalw=2.1=hc9558a2_5
+  - cmake=3.17.0=h28c56e5_0
```
Comment (apeltzer) [inline environment.yml:55]: Do we really need that one?

File: environment.yml [hunk 2]
```diff
@@ -6,36 +6,525 @@ channels:
...
+  - cnvkit=0.9.6=py27_1
+  - coloredlogs=14.0=py27h8c360ce_1
+  - colormath=3.0.0=py_2
+  - control-freec=11.6=he1b5a44_0
+  - cryptography=2.8=py27h2c19f6a_2
```
Comment (apeltzer) [inline environment.yml:60]: Do we really need that one?

File: environment.yml [hunk 3]
```diff
@@ -6,36 +6,525 @@ channels:
...
+  - ensembl-vep=99.2=pl526hecc5488_0
+  - enum34=1.1.10=py27h8c360ce_1
+  - fastqc=0.11.9=0
+  - fgbio=1.1.0=0
+  - font-ttf-dejavu-sans-mono=2.37=hab24e00_0
+  - font-ttf-inconsolata=2.001=hab24e00_0
+  - font-ttf-source-code-pro=2.030=hab24e00_0
+  - font-ttf-ubuntu=0.83=hab24e00_0
+  - fontconfig=2.13.1=hba837de_1004
+  - fonts-conda-ecosystem=1=0
```
Comment (apeltzer) [inline environment.yml:74]: Thats quite a lot of fonts actually. For which bits are these required ...? The FastQC recipe was fixed at some point by Devon & Me, so shouldn't require any of these. Might be that other of your packages need it but I can't tell :-(

**Review by apeltzer — COMMENTED** [review]
Its quite a long list now - really focus on the major ones and hope that it resolves it properly. Also I heard mixed experiences with having R packages that are not from conda-forge + bioconda but also from other R channels to be largely incompatible with these :-(

**Comment by MaxUlysse** [comment]
I actually have no idea, I just copied the environment that was successful in mamba here.
I was happy that it worked already

File: environment.yml (hunk 2 above)
Comment (MaxUlysse) [reply environment.yml:60]: this one yes: `control-freec=11.6=he1b5a44_0`

**Comment by MaxUlysse** [comment]
Ok, so I'll try to remove as much as possible and see if it still works as well.
Thanks for the tips

**Comment by apeltzer** [comment]
Yes, I think that would make sense 👍🏼 Otherwise updating this will be extremely difficult :-(

**Comment by MaxUlysse** [comment]
> Yes, I think that would make sense 👍🏼 Otherwise updating this will be extremely difficult :-(

I don't plan on updating this ever actually.
Plan is to get rid of it once 2.7 is done and go full dsl2 for 3.0

**Review by FriederikeHanssen — APPROVED** [review]

**Comment by apeltzer** [comment]
Much better :-)

**Comment by MaxUlysse** [comment]
Definitively better, but in the end, I have no idea why it's working when the previous version was not.

**Comment by apeltzer** [comment]
Mistery conda ;-)

**Pull request merged**